src/modules/azure-azcopy-sync.ts

```diff
@@ -0,0 +1,134 @@
+import { Echo } from "./echo";
+import shell from "shelljs";
+import { CommandStringBuilder } from "../utilities/command-string-builder";
+import child_process from "child_process";
+
+// -----------------------------------------------------------------------------------------
+// #region Constants
+// -----------------------------------------------------------------------------------------
+
+const AZCOPY_COMMAND = "azcopy sync";
+
+// #endregion Constants
+
+// -----------------------------------------------------------------------------------------
+// #region Interfaces
+// -----------------------------------------------------------------------------------------
+
+interface BlobStorageUrlParts {
+    account: string;
+    container: string;
+    path?: string;
+    sasToken?: string;
+}
+
+interface SyncLocalFileOptions {
+    localFilePath: string;
+    destination: BlobStorageUrlParts;
+}
+
+interface SyncContainerOptions {
+    deleteDestination: boolean;
+    destination: BlobStorageUrlParts;
+    recursive: boolean;
+    source: BlobStorageUrlParts;
+}
+
+// #endregion Interfaces
+
+// -----------------------------------------------------------------------------------------
+// #region Functions
+// -----------------------------------------------------------------------------------------
+
+const _getCommandStringBuilder = (
+    source: string,
+    destination: string,
+    flags: string[]
+): CommandStringBuilder => {
+    return new CommandStringBuilder(
+        "azcopy",
+        "sync",
+        source,
+        destination,
+        ...flags
+    );
+};
+
+const _getBlobStorageUrl = (urlParts: BlobStorageUrlParts) => {
+    const path = urlParts.path ?? "";
+    const sasToken = urlParts.sasToken ?? "";
+    return `"https://${urlParts.account}.blob.core.windows.net/${urlParts.container}/${path}${sasToken}"`;
+};
+
+const _getDeleteDestination = (deleteDestination: boolean) => {
```

Comment:
Nitpick: Consider renaming to `_getDeleteDestinationFlag` to be consistent with the other `get...Flag()` method below.